zerver/views/users.py

```diff
@@ -173,7 +173,10 @@ def patch_bot_backend(
     if full_name is not None:
         check_change_full_name(bot, full_name, user_profile)
     if bot_owner is not None:
-        owner = get_user(bot_owner, user_profile.realm)
+        try:
+            owner = get_user(bot_owner, user_profile.realm)
+        except UserProfile.DoesNotExist:
+            return json_error(_('Failed to change owner, no such user exist'))
```

Comment:
I think this should say: 'Failed to change owner; no such user' (with the `exist`) to match the error for the bot user themself not existing above.  

Also, this should be a second commit since it's an unrelated bug, but we need to add a check here: we should verify the new `bot_owner` is an active, non-bot user.  

Otherwise this PR looks good; happy to merge once this is resolved.